pkgs/games/heroic/default.nix

```diff
@@ -88,9 +90,19 @@ stdenv.mkDerivation (finalAttrs: {
 
     substituteInPlace "$out/share/heroic/flatpak/com.heroicgameslauncher.hgl.desktop" \
       --replace-fail "Exec=heroic-run" "Exec=heroic"
-    mkdir -p "$out/share/applications" "$out/share/icons/hicolor/512x512/apps"
+    mkdir -p "$out/share/applications" "$out/share/icons/hicolor/scalable/apps"
     ln -s "$out/share/heroic/flatpak/com.heroicgameslauncher.hgl.desktop" "$out/share/applications"
-    ln -s "$out/share/heroic/flatpak/com.heroicgameslauncher.hgl.png" "$out/share/icons/hicolor/512x512/apps"
+    install -Dm644 $out/share/heroic/src/frontend/assets/heroic-icon.svg $out/share/icons/hicolor/scalable/apps/com.heroicgameslauncher.hgl.svg
+    for size in 16 22 32 48 64 128 256 512; do
+      dir="$out/share/icons/hicolor/''${size}x''${size}/apps"
+      mkdir -p $dir
+      rsvg-convert \
+        --keep-aspect-ratio \
+        --width $size \
+        --height $size \
+        --output $dir/com.heroicgameslauncher.hgl.png \
+        $out/share/icons/hicolor/scalable/apps/com.heroicgameslauncher.hgl.svg
+    done
```

Comment:
I don't think this is needed if we already included the svg. SVG as desktop icon is part of the freedesktop specification, and can satisfy every display size.

Reply:
I did the modifications, but I wasn't able to test it yet.